File: src/main/java/org/taktik/icure/logic/MedicalLocationLogic.java
package org.taktik.icure.logic;

import org.taktik.icure.entities.MedicalLocation;
import org.taktik.icure.exceptions.DeletionException;

import java.util.List;

public interface MedicalLocationLogic extends EntityPersister<MedicalLocation, String> {

    MedicalLocation createMedicalLocation(MedicalLocation medicalLocation);

    List<String> deleteMedicalLocation(List<String> ids) throws DeletionException;

    MedicalLocation getMedicalLocation(String medicalLocation);

    MedicalLocation modifyMedicalLocation(MedicalLocation medicalLocation);

    List<MedicalLocation> findByPostCode(String postCode);
}

File: src/main/kotlin/org/taktik/icure/asynclogic/MedicalLocationLogic.kt
package org.taktik.icure.asynclogic

import kotlinx.coroutines.flow.Flow
import org.taktik.couchdb.DocIdentifier
import org.taktik.icure.entities.MedicalLocation

interface MedicalLocationLogic : EntityPersister<MedicalLocation, String> {
    suspend fun createMedicalLocation(medicalLocation: MedicalLocation): MedicalLocation?
    fun deleteMedicalLocations(ids: List<String>): Flow<DocIdentifier>

    suspend fun getMedicalLocation(medicalLocation: String): MedicalLocation?

    suspend fun modifyMedicalLocation(medicalLocation: MedicalLocation): MedicalLocation?
    fun findByPostCode(postCode: String): Flow<MedicalLocation>
}
